scripts/check-docker-pin.sh

```diff
@@ -0,0 +1,14 @@
+# This script is checks to that all our Docker images are pinned to a specific SHA256 hash
+#
+# References as to why...
+#   - https://nickjanetakis.com/blog/docker-tip-18-please-pin-your-docker-image-versions
+#   - https://snyk.io/blog/10-docker-image-security-best-practices/ (Specifically: USE FIXED TAGS FOR IMMUTABILITY)
+#
+find ~+ -type f -name "Docker*" -print0 | xargs -0 grep -s "FROM" {} | grep -v scratch | grep -v sha256 | grep -v "solana AS builder" | grep -v "solana AS ci_tests" | grep -v "node_modules"
```

Comment:
You might change the `solana AS foo` lines with just `AS` since AS is different.

```suggestion
find ~+ -type f -name "Docker*" -print0 | xargs -0 grep -s "FROM" {} | egrep -v 'scratch|sha256|solana AS builder|solana AS ci_tests|node_module'
```

I'd probably do it more like this, which might be a better approach:

```
[jschroeder@gusc1a-ossdev-jsl3 wormhole]$ find . -name 'Dockerfile*' -print0  -type f | xargs -0 grep FROM | egrep -v 'sha256|as|AS|node_modules'
./third_party/redis/Dockerfile:FROM alpine:3.14
./algorand/sandbox-algorand/images/indexer/Dockerfile:FROM golang:$GO_VERSION-alpine
./algorand/sandbox-algorand/images/Dockerfile:FROM ubuntu:20.04
./algorand/Dockerfile:FROM docker.io/python:3.10
```